crates/uplc/src/machine.rs

```diff
@@ -237,10 +237,10 @@ impl Machine {
             unspent_step_budget.occurences(self.unbudgeted_steps[i] as i32);
 
             self.spend_budget(unspent_step_budget)?;
+            
```

Comment:
This update does not set self.unbudgeted_steps[7] to zero. So to do it this way we need a final step after the for loop to do that.